File: android/src/main/kotlin/dev/eaceto/mobile/oss/flutter/flutter_local_authentication/LocalizationModel.kt
package dev.eaceto.mobile.oss.flutter.flutter_local_authentication

data class LocalizationModel(
    val dialogTitle: String,
    val reason: String,
    val cancelButtonTitle: String,
) {
    companion object {
        val titleKey = "promptDialogTitle"
        val reasonKey = "promptDialogReason"
        val cancelKey = "cancelButtonTitle"

        val default = LocalizationModel(
            "Biometric Prompt",
            "Validate that you have access to this device.",
            "Cancel"
        )

        fun from(dictionary: Map<String, Any>?): LocalizationModel? {
            return dictionary?.let {
                val title = dictionary[titleKey] as? String ?: return null
                val reason = dictionary[reasonKey] as? String ?: return null
                val cancel = dictionary[cancelKey] as? String ?: return null
                return LocalizationModel(
                    title,
                    reason,
                    cancel
                )
            }
        }
    }
}

File: macos/Classes/LocalizationModel.swift
//
//  LocalizationModel.swift
//  flutter_local_authentication
//
//  Created by Ezequiel (Kimi) Aceto on 19/10/23.
//  Contact: ezequiel.aceto@gmail.com
//  WebSite: https://eaceto.dev

import Foundation

struct LocalizationModel {

    fileprivate static let titleKey = "promptDialogTitle"
    fileprivate static let reasonKey = "promptDialogReason"
    fileprivate static let cancelKey = "cancelButtonTitle"

    let dialogTitle: String
    let reason: String
    let cancelButtonTitle: String

    static var `default` = LocalizationModel(
        dialogTitle: "",
        reason: "Validate that you have access to this device.",
        cancelButtonTitle: "Cancel"
    )

    static func from(_ dictionary: [String: Any]? = nil) -> LocalizationModel? {
        guard let dictionary,
              let title = dictionary[titleKey] as? String,
              let reason = dictionary[reasonKey] as? String,
              let cancel = dictionary[cancelKey] as? String else {
            return nil
        }
        return LocalizationModel(dialogTitle: title, reason: reason, cancelButtonTitle: cancel)
    }
}
